### PR TITLE
Issue #2187 - Support compartment search via POST

### PR DESCRIPTION
**Pull request by tbieste**
Issue #2187 - Support compartment search via POST

Signed-off-by: Troy Biesterfeld <tbieste@us.ibm.com>

### CODE REVIEW

**Review by michaelwschroeder — APPROVED** [review]
LGTM

File: fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java [hunk 1]
```diff
@@ -85,51 +96,27 @@ public Response search(@PathParam("type") String type) {
...
-            }
-
-            log.exiting(this.getClass().getName(), "search(String,String,String)");
-        }
+        return doSearchCompartment(compartment, compartmentId, type);
     }
 
     @POST
     @Consumes("application/x-www-form-urlencoded")
-    @Path("{type}/_search")
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java:130]: never mind, now i see it on new line 66

[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java:130]: its pretty interesting that the GET and POST variants are handled *exactly* the same.
what happens if someone happens to send a GET request with a non-empty body?  would we now process that similar to how we process it in the POST case?

File: fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java [hunk 2]
```diff
@@ -85,51 +96,27 @@ public Response search(@PathParam("type") String type) {
...
-    public Response _search(@PathParam("type") String type) {
-        log.entering(this.getClass().getName(), "_search(String)");
+    @Path("{compartment}/{compartmentId}/{type}/_search")
+    public Response searchCompartmentPost(@PathParam("compartment") String compartment,
+        @PathParam("compartmentId") String compartmentId, @PathParam("type") String type) {
```
Comment (lmsurpre) [inline fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java:114]: style comment
```suggestion
            @PathParam("compartmentId") String compartmentId, @PathParam("type") String type) {
```

File: fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbieste) [reply fhir-server/src/main/java/com/ibm/fhir/server/resources/Search.java:130]: The logic was already exactly the same, so I just combined them into a common method. For a GET request, uriInfo.getQueryParameters() does not return anything passed the request body, so looks like it is just completely ignored.

**Review by lmsurpre — APPROVED** [review]
LGTM

**Pull request merged**